### PR TITLE
21711: Playlist Detail

### PR DESCRIPTION
**Pull request by anhnv-0990**
## [Pull request title]
Playlist Detail
### [Redmine link]
21711: https://edu-redmine.sun-asterisk.vn/issues/21711
21712 UI : https://edu-redmine.sun-asterisk.vn/issues/21712
21713 Fetch data: https://edu-redmine.sun-asterisk.vn/issues/21713
#### Screenshots (nếu có)
Screen Shot 2020-03-18 at 1 17 56 PM
Screen Shot 2020-03-18 at 1 18 10 PM copy

### CODE REVIEW

File: Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+//MARK: - UITableViewDataSource
+extension PlaylistDetailViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return items.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(for: indexPath, cellType: TrackPlaylistDetailTableViewCell.self)
+        cell.detailTrack = items[indexPath.row]
```
Comment (datdt-1935) [inline Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift]:
```suggestion
        cell.configCell(track: items[indexPath.row])
```

File: Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    //MARK: - IBOutlet
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var ownerImage: UIImageView!
+    @IBOutlet private weak var ownerName: UILabel!
+    @IBOutlet private weak var descriptionsLabel: UILabel!
+    @IBOutlet private weak var timeLabel: UILabel!
+    @IBOutlet private weak var numberLikeLabel: UILabel!
+    
+    //MARK: - Detail View
+    private lazy var detailView = TopDetailView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: 0), with: playlist)
```
Comment (longvt-1468) [inline Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift]: break line

File: Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    //MARK: - Life Cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configView()
+        fetchUsersProfile(user: playlist.owner)
```
Comment (longvt-1468) [inline Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift]: tách 3 dòng này ra 1 hàm fetchData

File: Soundify/Soundify/Scenes/PlaylistDetail/TrackPlaylistDetailTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+//  Soundify
+//
+//  Created by Viet Anh on 3/16/20.
+//  Copyright © 2020 VietAnh. All rights reserved.
+//
+
+import UIKit
+import Reusable
+
+class TrackPlaylistDetailTableViewCell: UITableViewCell, Reusable {
```
Comment (longvt-1468) [inline Soundify/Soundify/Scenes/PlaylistDetail/TrackPlaylistDetailTableViewCell.swift]: add final

File: Soundify/Soundify/Scenes/YourLibrary/YourLibraryViewController.swift [hunk 5]
```diff
@@ -60,16 +61,29 @@ extension YourLibraryViewController: UITableViewDataSource {
...
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
         return Constants.TableView.heightForRow
     }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        switch indexPath.row {
+        case 0:
+            return
+        default:
+            guard let playlistDetailVC = AppStoryboard.playlistDetail.instantiateInitialViewController() as? PlaylistDetailViewController else { return }
```
Comment (longvt-1468) [inline Soundify/Soundify/Scenes/YourLibrary/YourLibraryViewController.swift]: break line

File: Soundify/Soundify/Data/Model/PlaylistDetail.swift [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    var name: String = ""
+    var owner: User
+    var snapshotId: String = ""
+    var type: String = ""
+    var uri: String = ""
+}
+
+struct PlaylistDetailTrack: BaseModel {
+    var track: Track
+}
```
Comment (dongtg-1485) [inline Soundify/Soundify/Data/Model/PlaylistDetail.swift]: tách file

File: Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+import UIKit
+import Reusable
+
+final class PlaylistDetailViewController: BaseDetailViewController {
+    
+    //MARK: - IBOutlet
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var ownerImage: UIImageView!
+    @IBOutlet private weak var ownerName: UILabel!
```
Comment (dongtg-1485) [inline Soundify/Soundify/Scenes/PlaylistDetail/PlaylistDetailViewController.swift]:
```suggestion
    @IBOutlet private weak var ownerImageView: UIImageView!
    @IBOutlet private weak var ownerNameLabel: UILabel!
```

**Review by dongtg-1485 — APPROVED** [review]

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**